### PR TITLE
G-API Fixed fluid roi inference to always produce even sizes for nv12

### PR DESCRIPTION
**Pull request by rgarnov**
* Added an adjustment for roi produced by NV12toRGB to always be even

### CODE REVIEW

**Comment by rgarnov** [comment]
@andrey-golubev please take a look

File: modules/gapi/src/backends/fluid/gfluidbackend.cpp [hunk 1]
```diff
@@ -658,7 +658,22 @@ void cv::gimpl::GFluidExecutable::initBufferRois(std::vector<int>& readStarts,
...
 
                     int readStart = resized.y;
-                    cv::gapi::own::Rect roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+                    auto roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+
+                    // Extend required input roi to be even if it's produced by NV12toRGB
+                    if (!in_node->inNodes().empty()) {
+                        auto in_data_producer = in_node->inNodes().front();
+                        if (fg.metadata(in_data_producer).get<FluidUnit>().k.m_kind == GFluidKernel::Kind::NV12toRGB) {
+                            if (roi.y % 2 != 0) {
```
Comment (alalek) [inline modules/gapi/src/backends/fluid/gfluidbackend.cpp]: What should be even?
- base `.y`
- or `.height`

---

What is about `.width`?

File: modules/gapi/src/backends/fluid/gfluidbackend.cpp [hunk 2]
```diff
@@ -658,7 +658,22 @@ void cv::gimpl::GFluidExecutable::initBufferRois(std::vector<int>& readStarts,
...
+                                roi.y--;
+                                roi.height++;
+                            }
+
+                            if (roi.height % 2 != 0) {
+                                roi.height++;
```
Comment (alalek) [inline modules/gapi/src/backends/fluid/gfluidbackend.cpp]: What is about a check with fitting into image?
Why is not to shrink `roi`?

File: modules/gapi/src/backends/fluid/gfluidbackend.cpp (hunk 1 above)
Comment (rgarnov) [reply modules/gapi/src/backends/fluid/gfluidbackend.cpp]: Both .y and .height should be even, vertical slicing is not supported for now, so width is not taken into account

File: modules/gapi/src/backends/fluid/gfluidbackend.cpp (hunk 2 above)
Comment (rgarnov) [reply modules/gapi/src/backends/fluid/gfluidbackend.cpp]: This is actually done later via adjFilterRoi call, I was thinking to clarify this step, will do

**Pull request merged**